modules/delegates.js

```diff
@@ -429,69 +423,68 @@ __private.checkDelegates = function(senderPublicKey, votes, state, cb, tx) {
 
 						if (
 							!votesAccounts ||
-							votesAccounts.length < voteAddressesWithActions.length
+							votesAccounts.length < votesWithAction.length
 						) {
 							library.logger.error(
 								'Delegates with addresses not found',
 								_.differenceWith(
-									voteAddressesWithActions,
+									votesWithAction,
 									votesAccounts,
 									(
-										{ address: addressFromTransaction },
-										{ addressFromAccount }
-									) => addressFromTransaction === addressFromAccount
-								).map(({ address }) => address)
+										{ publicKey: publicKeyFromTransaction },
+										{ publicKey: publicKeyFromAccount }
+									) => publicKeyFromTransaction === publicKeyFromAccount
+								).map(({ publicKey }) => publicKey)
 							);
 							return setImmediate(waterfallCb, 'Delegate not found');
 						}
-
 						const upvoteAccounts = votesAccounts.filter(voteAccount =>
-							voteAddressesWithActions.find(
-								actionWithAddress =>
-									actionWithAddress.action === '+' &&
-									actionWithAddress.address === voteAccount.address
+							votesWithAction.find(
+								({ action, publicKey }) =>
+									action === '+' && publicKey === voteAccount.publicKey
 							)
 						);
 						const downvoteAccounts = votesAccounts.filter(voteAccount =>
-							voteAddressesWithActions.find(
-								actionWithAddress =>
-									actionWithAddress.action === '-' &&
-									actionWithAddress.address === voteAccount.address
+							votesWithAction.find(
+								({ action, publicKey }) =>
+									action === '-' && publicKey === voteAccount.publicKey
 							)
 						);
 
-						const invalidUpvoteAccount = upvoteAccounts.find(
-							voteAccount =>
-								existingVotedAddresses.find(
-									existingVotedAddress =>
-										voteAccount.address === existingVotedAddress
-								) !== undefined
+						const invalidUpvoteAccounts = _.intersectionWith(
+							upvoteAccounts,
+							existingVotedPublicKeys,
+							({ publicKey: upvoteAccountPublicKey }, existingVotedPublicKey) =>
+								upvoteAccountPublicKey === existingVotedPublicKey
 						);
```

Comment:
Can be simplified as,

```js
const invalidUpvoteAccounts = _.intersect(upvotePublicKeys, existingVotedPublicKeys);
```

Reply:
Doing this because I don't want to traverse the list again in case of invalidUpvoteAccount and invalidDownvoteAccount checks.